spec/classes/init_spec.rb

```diff
@@ -16,7 +16,11 @@
         it { is_expected.to contain_exec('generate_zram_units').with_refreshonly(true) }
         it { is_expected.to contain_file('/usr/lib/systemd/zram-generator.conf.d').with_ensure('directory') }
         it { is_expected.to contain_package('zram-generator-defaults').with_ensure('absent') }
-        it { is_expected.to contain_package('zram-generator').with_ensure('installed') }
+        if facts [:os]['name'] == 'Archlinux'
+          it { is_expected.not_to contain_package('zram-generator')
+        else
+          it { is_expected.to contain_package('zram-generator').with_ensure('installed') }
```

Comment:
`zram-generator` should still be present - it's the defaults (a line above) that you need to change.